### PR TITLE
drivers: fix G305 quirk detection

### PR DESCRIPTION
**Pull request by crab2313**
Fixes: #833 
Fixes: #846 

The second patch is not well tested. But I think it should work. Linux kernel version 5.4 causes serious regression of my system and force me to switch back to an old kernel.

### CODE REVIEW

File: src/driver-hidpp20.c [hunk 1]
```diff
@@ -1509,7 +1509,6 @@ hidpp20drv_remove(struct ratbag_device *device)
 	if (drv_data->profiles)
 		hidpp20_onboard_profiles_destroy(drv_data->profiles);
 	free(drv_data->led_infos.color_leds_8070);
-	free(drv_data->led_infos.color_leds_8071);
 	free(drv_data->controls);
```
Comment (whot) [inline src/driver-hidpp20.c]: whoah, yeah, that's a double free, that info is in a union. Though really we need to wrap the `free()` calls into a capability check, similar to `hidpp20drv_read_color_leds()`.

File: src/hidpp20.c [hunk 2]
```diff
@@ -2627,7 +2627,7 @@ hidpp20_onboard_profiles_initialize(struct hidpp20_device *device,
 						  profiles->sector_size,
 						  data);
 
-	if (rc && device->quirk != HIDPP20_QUIRK_G305) {
+	if (rc && device->quirk == HIDPP20_QUIRK_G305) {
```
Comment (whot) [inline src/hidpp20.c]: @FFY00 - typo, right? this patch is correct?

Comment (FFY00) [reply src/hidpp20.c]: Yes!

File: src/driver-hidpp20.c (hunk 1 above)
Comment (FFY00) [reply src/driver-hidpp20.c]: We can make a macro to free and make the value NULL.

Comment (crab2313) [reply src/driver-hidpp20.c]: @FFY00  I will remove the second commit if you decide to fix it later.

Comment (whot) [reply src/driver-hidpp20.c]: > We can make a macro to free and make the value NULL.

we'd still rely on this struct to be free-able with a single call. better to be correct here and add the `if` so we don't introduce new bugs if we ever extend this.

**Comment by FFY00** [comment]
Can you split this in two commits? One fixing the G305 quirk and one fixing the double free. Also, if you could use the `hidpp20` prefix instead of `drivers` it would be great!

**Comment by crab2313** [comment]
@FFY00 Already two commits.

**Pull request merged**